### PR TITLE
VS-13639 adds label validation feature to google cloud vision requests

### PR DESCRIPTION
**Pull request by jpcfernandes**
Introduces label validation. API responses are now wrapped on a VisionResponse object.

### CODE REVIEW

File: src/VisionClient.php [hunk 1]
```diff
@@ -6,70 +6,80 @@
...
+
+        $this->api_key = $api_key;
+    }
+
+    /**
+     * @return Client
+     */
+    protected static function getClient()
+    {
+        return new Client([
```
Comment (joecollier) [inline src/VisionClient.php]: InvalidArgumentException is currently being thrown: https://rpm.newrelic.com/accounts/315565/applications/22651594/traced_errors/fb23f827-a9af-11e7-8fac-0242ac110008_9257_12483?original_error_id=1fd2e5e9-a9c3-11e7-8fac-0242ac110008_9380_12607

Is there anything we should be doing in the VisionClient to mitigate this issue?

Comment (artur-dev) [reply src/VisionClient.php]: Good question, looks like referer is empty in some cases

File: src/VisionRequest.php [hunk 2]
```diff
@@ -42,162 +47,182 @@ class VisionRequest
...
+     * @param string $image_send_mode
+     * @return VisionRequest
+     * @throws VisionRequestException
      */
-    public function setImageRequestType($image_request_type)
+    public function setImageSendMode($image_send_mode)
     {
-        $this->image_request_type = $image_request_type;
+        if (!in_array($image_send_mode, static::getAvailableSendModes())) {
+            throw new VisionRequestException("Unavailable image send mode: 'image_send_mode''");
```
Comment (artur-dev) [inline src/VisionRequest.php]: were you going to print actual $image_send_mode or just 'image_send_mode' string?
BTW there is repeated single quote after image_send_mode

File: src/VisionRequest.php [hunk 3]
```diff
@@ -42,162 +47,182 @@ class VisionRequest
...
-            ]
-        ];
-    }
+            ];
+        } elseif ($this->getImageSendMode() == static::SEND_IMAGE_CONTENT_MODE) {
+            $image_params = [
+                'content' => static::getImageContent($this->getImageUri())
+            ];
+        } else {
+            throw(new VisionRequestException('Image send mode not set'));
```
Comment (artur-dev) [inline src/VisionRequest.php]: why extra brackets after throw?

File: src/VisionRequest.php (hunk 2 above)
Comment (jpcfernandes) [reply src/VisionRequest.php]: 👍

File: src/VisionRequest.php (hunk 3 above)
Comment (jpcfernandes) [reply src/VisionRequest.php]: 👍

**Comment by mcinbell** [comment]
retest this please

**Comment by mcinbell** [comment]
retest this please

**Comment by mcinbell** [comment]
retest this please

**Comment by mcinbell** [comment]
retest this please

**Comment by mcinbell** [comment]
retest this please

**Comment by mcinbell** [comment]
retest this please

**Pull request merged**